lc-hax/Scripts/Patches/ItemsPatch.cs

```diff
@@ -0,0 +1,12 @@
+#pragma warning disable IDE1006
+
+using HarmonyLib;
+
+[HarmonyPatch(typeof(GrabbableObject), nameof(GrabbableObject.Start))]
+class ItemsPatch {
+    static void Postfix(Item? ___itemProperties) {
+        if (___itemProperties == null) return;
+        ___itemProperties.canBeGrabbedBeforeGameStart = true;
+        ___itemProperties.twoHanded = false;
```

Comment:
This doesn't work with `LungProp` aka the Apparatus.

Reply:
Ah okay, I remember now. So if you inject when you are already in lobby, this won't work. It's because `Start` has already been called.

Reply:
Technically you can do a check on mod launch and edit all the grabbable objects again and re-adjust their respective settings.

Reply:
I thought of that as well but then our solution has now become more expensive and complex than setting a bunch of fields on LateUpdate.

Reply:
> Owner

it needs to be done once, not all the times, because harmony will take care of it .

Reply:
Sure but you are also paying that steep initialisation cost where we loop through every item in the game and reinitialise them.

Reply:
I am not sure what you mean. We are currently patching `Start`. The only way to make this work is to call the `Start` method on every item.

Reply:
> I am not sure what you mean. We are currently patching `Start`. The only way to make this work is to call the `Start` method on every item.

No, dont do that, just edit the itemProperty of each item without calling start, you can easily do it with a foreach loop once, and forget about it if the mod is injected during gameplay, else harmony will take care of it, nothing is changing the ItemProperties fields, is in the prefab itself, which will never be edited, i checked it.

Reply:
Yeah, that sounds like additional complexity. Effectively, we would have two methods doing the same work. The initialiser and the patch (for new item spawns).

Reply:
> Yeah, that sounds like additional complexity. Effectively, we would have two methods doing the same work. The initialiser and the patch (for new item spawns).

initializer is a failsafe for doing the same thing the patch is doing, which is modifying every ItemProperty in each spawned items if the harmony patch missed these ( the mod gets injected during the game session ), then yes, it will do the same edit, but it will be ran once, then it will be discarded as harmony will do the work.

Reply:
I try to avoid patching game loops as well but in this case, the trade-off is worth it and the performance hit is nearly non-existent as we aren't calling any methods.

Reply:
> I try to avoid patching game loops as well but in this case, the trade-off is worth it and the performance hit is nearly non-existent as we aren't calling any methods.

also the "inventory full" alert is no longer showing up with this patch.

Reply:
I am still getting it though?

![image](https://github.com/winstxnhdw/lc-hax/assets/56998716/e4e0f70a-bc0a-406b-8fdd-b9ecca58dd56)

Reply:
> I am still getting it though?
> 
> ![image](https://private-user-images.githubusercontent.com/56998716/298218798-e4e0f70a-bc0a-406b-8fdd-b9ecca58dd56.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU3MDQ4NzcsIm5iZiI6MTcwNTcwNDU3NywicGF0aCI6Ii81Njk5ODcxNi8yOTgyMTg3OTgtZTRlMGY3MGEtYmMwYS00MDZiLThmZGQtYjllY2NhNThkZDU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDIyNDkzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwNGY1ZjI2ZmI4YzVhOWNhOTlhYmQxNzQyNDQ5MmZkNGY0Mzg4MDM4ZGQyYTdlYjNkMmFmZTIxNTQzZjc1YmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IOJ_Y3PfFr4CWLePo5OFanl2kxPc_08yvIRgapQxwj4)

Because your inventory actually is full, i mean when you hold a two handed item LOL

Reply:
Ahh, you meant that. Actually, that was intended and you just reminded me why I chose to do it the previous way. IIRC the previous way spoofed your two-handed animation to other players and prevented them from realising that you are actually holding the item one-handed.

Reply:
> Ahh, you meant that. Actually, that was intended and you just reminded me why I chose to do it the previous way. IIRC the previous way spoofed your two-handed animation to other players and prevented them from realising that you are actually holding the item one-handed.

oh, i did not though of that, then scrap the way im doing it, but you can keep the patch to allow people to grab ship's items on join instead of waiting for it to land.

Reply:
Yeap, I'll be sure to keep that.